src/librbd/librbd.cc

```diff
@@ -3126,6 +3175,25 @@ namespace librbd {
     ictx->state->quiesce_complete(handle, r);
   }
 
+  int Image::get_pool_name(std::string &pool_name)
+  {
+    pool_name = reinterpret_cast<ImageCtx *>(ctx)->md_ctx.get_pool_name();
+    return 0;
+  }
+
+  int Image::get_snap_name(snap_t in_snap_id, string &out_snap_name)
+  {
+    ImageCtx *ictx = (ImageCtx *)ctx;
+
+    std::shared_lock image_locker{ictx->image_lock};
+    int r = ictx->get_snap_name(in_snap_id, &out_snap_name);
+    if (r < 0) {
+      return r;
+    }
+
+    return r;
```

Comment:
why not just return `r`?

Reply:
I would prefer `return 0;`. Our convention is:
```
r = ...
if (r < 0) {
  ...
  return r;
}
...
return 0;
```

Reply:
@tchaikov @trociny The code style has been modified